src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolver.vsdconfigxml

```diff
@@ -0,0 +1,32 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--
+
+     Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+-->
+<Configuration xmlns="http://schemas.microsoft.com/vstudio/vsdconfig/2008">
+  <DefineGuid Name="CSharpFunctionResolverId" Value="CFE62047-C343-4DE4-B31E-88B7AFF8E7F7"/>
+  <ManagedComponent
+    ComponentId="CSharpFunctionResolverId"
+    ComponentLevel="65100"
```

Comment:
Good catch!
Yes, you'll want to make this Synchronized as well as implement OnModuleModified and OnModuleInstanceUnload.  If don't make it synchronized, you'll need to lock around all metadata access and ModuleInstance Load/Unload/Modify.